repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/PreProcessorUtils.java

```diff
@@ -198,4 +212,208 @@ public static void verifyDuplicateAssetByName(String typeName, String assetName,
+            }
+        }
+
+        LexoRank parsedLexoRank = LexoRank.parse(lastLexoRank);
+        LexoRank nextLexoRank = parsedLexoRank.genNext().genNext();
+        lexoRank = nextLexoRank.toString();
+
+        entity.setAttribute(LEXICOGRAPHICAL_SORT_ORDER, lexoRank);
+        lexoRankCache.put(entity.getTypeName() + "-" + glossaryQualifiedName + "-" + parentQualifiedName, lexoRank);
+        RequestContext.get().setLexoRankCache(lexoRankCache);
+    }
+
+    public static Map<String, Object> createDSLforCheckingPreExistingLexoRank(String lexoRank, String glossaryQualifiedName, String parentQualifiedName) {
+
+        Map<String, Object> sortKeyOrder = mapOf(LEXICOGRAPHICAL_SORT_ORDER, mapOf("order", "desc"));
+        Map<String, Object> scoreSortOrder = mapOf("_score", mapOf("order", "desc"));
+        Map<String, Object> displayNameSortOrder = mapOf("displayName.keyword", mapOf("order", "desc"));
+
+        Object[] sortArray = {sortKeyOrder, scoreSortOrder, displayNameSortOrder};
+
+        Map<String, Object> functionScore = mapOf("query", buildBoolQueryDuplicateLexoRank(lexoRank, glossaryQualifiedName, parentQualifiedName));
+
+        Map<String, Object> dsl = new HashMap<>();
+        dsl.put("from", 0);
+        dsl.put("size", 100);
+        dsl.put("sort", sortArray);
+        dsl.put("query", mapOf("function_score", functionScore));
+
+        return dsl;
+    }
+
+    private static Map<String, Object> buildBoolQueryDuplicateLexoRank(String lexoRank, String glossaryQualifiedName, String parentQualifiedName) {
+        Map<String, Object> boolQuery = new HashMap<>();
+        Map<String, Object> boolFilter = new HashMap<>();
+        List<Map<String, Object>> mustArray = new ArrayList<>();
+        List<Map<String, Object>> mustNotArray = new ArrayList<>();
+        mustArray.add(mapOf("term", mapOf("__state", "ACTIVE")));
+        mustArray.add(mapOf("term", mapOf(LEXICOGRAPHICAL_SORT_ORDER, lexoRank)));
+        if(StringUtils.isNotEmpty(glossaryQualifiedName)) {
+            mustArray.add(mapOf("terms", mapOf("__typeName.keyword", Arrays.asList("AtlasGlossaryTerm", "AtlasGlossaryCategory"))));
+            mustArray.add(mapOf("term", mapOf("__glossary", glossaryQualifiedName)));
+        } else{
+            mustArray.add(mapOf("terms", mapOf("__typeName.keyword", Arrays.asList("AtlasGlossary"))));
+        }
+
+        if(StringUtils.isEmpty(parentQualifiedName)) {
+            mustNotArray.add(mapOf("exists", mapOf("field", "__categories")));
+            mustNotArray.add(mapOf("exists", mapOf("field", "__parentCategory")));
+            boolFilter.put("must_not", mustNotArray);
+        }
+        else {
+            List<Map<String, Object>> shouldParentArray = new ArrayList<>();
+            shouldParentArray.add(mapOf("term", mapOf("__categories", parentQualifiedName)));
+            shouldParentArray.add(mapOf("term", mapOf("__parentCategory", parentQualifiedName)));
```

Comment:
Instead of adding both, please add which is relevant & add this into a must, e.g. in case for Term we should only add check for __categories & not __parentCategory